docs/components.rst

```diff
@@ -0,0 +1,76 @@
+
+.. code-block:: python
+
+    [...]
+    action_row = manage_components.create_actionrow(*buttons)
+
+
+Fantastic, we now have an action row with a green button in it, now lets get it sent in discord
+
+.. code-block:: python
+
+    await ctx.send("My Message", components=[action_row])
+
+Now if you've followed along, you have a green button in discord! But theres a problem, whenever you click it you see that the ``interaction failed``. Why is that?
+Well, in Discord, clicking buttons and using slash commands are called ``interactions``, and Discord doesn't know if we've received them or not unless we tell Discord. So how do we do that?
+
+Responding to interactions
+__________________________
+
+When responding to interactions, we have 2 choices, either to do it in the command itself, or a global event handler (like :meth:`on_slash_command_error`)
+
+Lets go through the most common method first, responding in the event itself. We simply need to :meth:`wait_for` the event, just like you do for reactions. For this we're going to use :meth:`wait_for_component() <discord_slash.utils.manage_components>`, and we're going to only wait for events from the action row we just .
+This method will return a :class:`ComponentContext <discord_slash.context.ComponentContext>` object that we can use to respond. For this example, we'll just edit the original message (:meth:`edit_origin() <discord_slash.context.ComponentContext.edit_origin>`)
+
+.. code-block:: python
+
+    await ctx.send("My Message", components=[action_row])
+    # note: this will only catch one button press, if you want more, put this in a loop
+    button_ctx: ComponentContext = await manage_components.wait_for_component(bot, action_row)
+    await button_ctx.edit_origin(content="You pressed a button!")
+
+.. note:: It's worth being aware that if you handle the event in the command itself, it will not persist reboots. As such when you restart the bot, the interaction will fail
+
+Next we'll go over the alternative, a global event handler. This works just the same as :meth:`on_slash_command_error` or `on_ready`.
+
+.. code-block:: python
+
+    @bot.event
+    async def on_component(ctx: ComponentContext):
+        await button_ctx.edit_origin(content="You pressed a button!")
+
+But [writer], I dont want to edit the message
+*********************************************
+
+Well lucky for you, you don't have to. You can either respond silently, with a thinking animation, or send a whole new message. Take a look here: :class:`ComponentContext <discord_slash.context.ComponentContext>`
+
+How do i know which button was pressed?
+_______________________________________
+
+Each button gets a ``custom_id``, this is a unique identifier of which button is being pressed. You can specify what the ID is when you define your button. When handling the event, simply check the custom_id, and handle accordingly.
```

Comment:
Worth mentioning that `create_button` generates random `custom_id` if it wasn't specified, and mention `get_components_ids` funtion to extract `custom_ids` from actionrow or list of actionrows

Reply:
I was trying to keep this as simple as possible, but i can add a note to explain that if you dont specify a `custom_id` a random one is given. I think mentioning `get_component_ids` is unnecessary in the scope of this guide however

Reply:
Ive added the note about `custom_id` random gen, not resolving due to the unresolved `get_components_ids` part